src/datachain/query/dataset.py

```diff
@@ -1097,9 +1103,26 @@ def __init__(
         self.column_types: Optional[dict[str, Any]] = None
 
         self.name = name
-        ds = self.catalog.get_dataset(name)
-        self.version = version or ds.latest_version
-        self.feature_schema = ds.get_version(self.version).feature_schema
+        try:
+            ds = self.catalog.get_dataset(name)
+            self.version = version or ds.latest_version
+            self.feature_schema = ds.get_version(self.version).feature_schema
+
+        except (DatasetNotFoundError, DatasetVersionNotFoundError):
+            if not studio:
+                raise
+
+            token = os.environ.get("DVC_STUDIO_TOKEN") or Config().read().get(
```

Comment:
can we move it somewhere / make a helper? it seems this should be a general logic that is used in many other places?